core/src/test/java/org/apache/iceberg/TestPartitionSpecBuilderCaseSensitivity.java

```diff
@@ -0,0 +1,993 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import static org.apache.iceberg.types.Types.NestedField.required;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.types.Types.NestedField;
+import org.apache.iceberg.types.Types.StructType;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+public class TestPartitionSpecBuilderCaseSensitivity {
+
+  private static final int V2_FORMAT_VERSION = 2;
+  private static final Schema SCHEMA_CASE_INSENSITIVE =
+      new Schema(
+          required(1, "id", Types.IntegerType.get()),
+          required(2, "data", Types.StringType.get()),
+          required(3, "category", Types.StringType.get()),
+          required(4, "order_date", Types.DateType.get()),
+          required(5, "order_time", Types.TimestampType.withoutZone()),
+          required(6, "ship_date", Types.DateType.get()),
+          required(7, "ship_time", Types.TimestampType.withoutZone()));
+
+  private static final Schema SCHEMA_CASE_SENSITIVE =
```

Comment:
I don't think this is correct. Schemas are neither case sensitive or case insensitive. They can be used in a case sensitive or insensitive context.

Reply:
Instead, how about with or without name conflicts?

Reply:
These constants have been renamed to SCHEMA_WITHOUT_NAME_CONFLICTS and SCHEMA_WITH_NAME_CONFLICTS